### DISCUSSION TITLE
How to obtain a generated parser's source code?

### DISCUSSION THREAD

**Discussion by danaugrs** [pest-parser/pest#588] [Q&A] [answered]
How would one go about getting the source code of a generated parser?

**Comment by glyn** [comment, marked as answer]
I used [`cargo expand`](https://github.com/dtolnay/cargo-expand). I put this in a [scratch file](https://wordsmith.social/underlap/git-scratch-directories) with extension `.rs` so my editor provides syntax highlighting:
```
$ cargo expand > scratch/generated.rs
```

If you only want documentation for the generated code, the pest book suggests `cargo doc` will do the trick, but I couldn't get that to work (even using the `--all` switch). I finally discovered that:
```
$ cargo doc --document-private-items --no-deps
```
documents the generated parser code.
